hil/cli.py

```diff
@@ -288,570 +194,194 @@ def serve_networks():
-
-
-@cmd
-def network_delete(network):
-    """Delete a <network>"""
-    C.network.delete(network)
-
-
-@cmd
-def user_delete(username):
-    """Delete the user <username>"""
-    C.user.delete(username)
-
-
-@cmd
-def list_projects():
-    """List all projects"""
-    q = C.project.list()
-    sys.stdout.write('%s Projects :    ' % len(q) + " ".join(q) + '\n')
-
-
-@cmd
-def user_add_project(user, project):
-    """Add <user> to <project>"""
-    C.user.add(user, project)
-
-
-@cmd
-def user_remove_project(user, project):
-    """Remove <user> from <project>"""
-    C.user.remove(user, project)
-
-
-@cmd
-def network_grant_project_access(project, network):
-    """Add <project> to <network> access"""
-    C.network.grant_access(project, network)
-
-
-@cmd
-def network_revoke_project_access(project, network):
-    """Remove <project> from <network> access"""
-    C.network.revoke_access(project, network)
-
-
-@cmd
-def project_create(project):
-    """Create a <project>"""
-    C.project.create(project)
-
-
-@cmd
-def project_delete(project):
-    """Delete <project>"""
-    C.project.delete(project)
-
-
-@cmd
-def headnode_create(headnode, project, base_img):
-    """Create a <headnode> in a <project> with <base_img>"""
-    url = object_url('headnode', headnode)
-    do_put(url, data={'project': project,
-                      'base_img': base_img})
-
-
-@cmd
-def headnode_delete(headnode):
-    """Delete <headnode>"""
-    url = object_url('headnode', headnode)
-    do_delete(url)
-
-
-@cmd
-def project_connect_node(project, node):
-    """Connect <node> to <project>"""
-    C.project.connect(project, node)
-
-
-@cmd
-def project_detach_node(project, node):
-    """Detach <node> from <project>"""
-    C.project.detach(project, node)
-
-
-@cmd
-def headnode_start(headnode):
-    """Start <headnode>"""
-    url = object_url('headnode', headnode, 'start')
-    do_post(url)
-
-
-@cmd
-def headnode_stop(headnode):
-    """Stop <headnode>"""
-    url = object_url('headnode', headnode, 'stop')
-    do_post(url)
+    C.node.set_bootdev(node, dev)
 
 
-@cmd
-def node_register(node, subtype, *args):
+@node.command(name='register', help='Register a node')
+@click.argument('node')
+@click.argument('obmtype')
+@click.argument('hostname')
+@click.argument('username')
+@click.argument('password')
+def node_register(node, obmtype, hostname, username, password):
     """Register a node named <node>, with the given type
         if obm is of type: ipmi then provide arguments
         "ipmi", <hostname>, <ipmi-username>, <ipmi-password>
     """
-    C.node.register(node, subtype, *args)
+    C.node.register(node, obmtype, hostname, username, password)
 
 
-@cmd
-def node_delete(node):
-    """Delete <node>"""
-    C.node.delete(node)
+###########################
+# node connect/detach network implementations. Could also be expanded to do
+# node connect/detach project if need be.
 
+@node.group(name='connect', help='Connect node to an <object>')
```

Comment:
Thinking we should make the `help` message consistent with each other. so maybe it is a good idea to change it to:

`Connect a node to an <object>`